src/adapter/src/catalog/consistency.rs

```diff
@@ -408,3 +555,42 @@ enum ObjectDependencyInconsistency {
         object_b: GlobalId,
     },
 }
+
+#[derive(Debug, Serialize, Clone, PartialEq)]
+enum ItemInconsistency {
+    /// The name in a `Database` `schemas_by_name` does not match the name on the `Schema` struct.
+    KeyedName {
+        db_schema_by_name: String,
+        struct_name: String,
+    },
```

Comment:
We should add similar checks for any of the following that don't already exist (Not necessarily in this PR):
- `database_by_name` and `database_by_id`
- `ambient_schemas_by_name` and `ambient_schemas_by_id`
- `clusters_by_name` and `clusters_by_id`
- `roles_by_name` and `roles_by_name`

Reply:
FWIW these mostly are already covered [here](https://github.com/MaterializeInc/materialize/blob/main/src/adapter/src/catalog/consistency.rs#L85-L124)